lib/auth/machineid/machineidv1/bot_service.go

```diff
@@ -0,0 +1,735 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package machineidv1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
+	pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/machineid/v1"
+	"github.com/gravitational/teleport/api/types"
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/events"
+	usagereporter "github.com/gravitational/teleport/lib/usagereporter/teleport"
+)
+
+// SupportedJoinMethods should match SupportedJoinMethods declared in
+// lib/tbot/config
+var SupportedJoinMethods = []types.JoinMethod{
+	types.JoinMethodAzure,
+	types.JoinMethodCircleCI,
+	types.JoinMethodGCP,
+	types.JoinMethodGitHub,
+	types.JoinMethodGitLab,
+	types.JoinMethodIAM,
+	types.JoinMethodKubernetes,
+	types.JoinMethodSpacelift,
+	types.JoinMethodToken,
+}
+
+// BotResourceName returns the default name for resources associated with the
+// given named bot.
+func BotResourceName(botName string) string {
+	return "bot-" + strings.ReplaceAll(botName, " ", "-")
```

Comment:
Do we need to remove `/` as well?

Reply:
At the moment, this has just moved the previous implementation of `BotResourceName` from `lib/auth` into `lib/auth/machineid/machineidv1` - I do agree that maybe we ought to replace `/` but I feel it's out of scope for this PR and I don't want to introduce additional risk by changing this.